tools/apiview/parsers/csharp-api-parser/CSharpAPIParser/Program.cs

```diff
@@ -204,6 +205,122 @@ static bool IsDll(string name)
         return name.EndsWith(".dll", StringComparison.OrdinalIgnoreCase);
     }
 
+    /// <summary>
+    /// Parses the target framework from a DLL path within a NuGet package.
+    /// </summary>
+    /// <param name="path">The path to the DLL within the package (e.g., "lib/net8.0/System.Text.Json.dll")</param>
+    /// <returns>The target framework moniker or null if not found</returns>
+    public static string? ParseTargetFrameworkFromPath(string path)
+    {
+        // Expected format: lib/{tfm}/{assembly}.dll or ref/{tfm}/{assembly}.dll
+        var parts = path.Split('/', StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length >= 3 && (parts[0] == "lib" || parts[0] == "ref"))
+        {
+            return parts[1];
+        }
+        return null;
+    }
+
+    /// <summary>
+    /// Selects the best target framework from the available options.
+    /// Prefers newer and more specific frameworks.
+    /// </summary>
+    /// <param name="availableFrameworks">Available target framework monikers</param>
+    /// <returns>The best target framework moniker</returns>
+    public static string? SelectBestTargetFramework(IEnumerable<string> availableFrameworks)
+    {
+        if (!availableFrameworks.Any()) return null;
+
+        var frameworks = availableFrameworks.Distinct().ToList();
+        
+        // Define framework preference order (higher number = better)
+        var frameworkPreferences = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "net462", 1 },
+            { "net47", 2 },
+            { "net471", 3 },
+            { "net472", 4 },
+            { "net48", 5 },
+            { "netstandard1.0", 10 },
+            { "netstandard1.1", 11 },
+            { "netstandard1.2", 12 },
+            { "netstandard1.3", 13 },
+            { "netstandard1.4", 14 },
+            { "netstandard1.5", 15 },
+            { "netstandard1.6", 16 },
+            { "netstandard2.0", 40 }, // Prefer netstandard2.0 over newer .NET versions for better compatibility
+            { "netstandard2.1", 25 },
+            { "netcoreapp1.0", 30 },
+            { "netcoreapp1.1", 31 },
+            { "netcoreapp2.0", 32 },
```

Comment:
I want to prefer NetStandard 2.0, then rest in the same order as you have, i.e. change the NetStandard 2.0 number to 30